src/rgw/rgw_op.cc

```diff
@@ -6394,10 +6394,11 @@ void RGWGetHealthCheck::execute()
 
 int RGWDeleteMultiObj::verify_permission()
 {
+  bool is_versioned = !rgw::sal::RGWObject::empty(s->object.get()) && s->object->have_instance();
   if (s->iam_policy || ! s->iam_user_policies.empty()) {
     auto usr_policy_res = eval_user_policies(s->iam_user_policies, s->env,
                                               boost::none,
-                                              s->object->get_instance().empty() ?
+                                              is_versioned ?
```

Comment:
Here, you have the logic the other way round from the previous version. Previously, if the object was not set, we would check the `s3DeleteObject` permission, and otherwise check `s3DeleteObjectVersion`. With the proposed change, if the object is not set, we'll check `s3DeleteObjectVersion`. Is that correct?